google/gax/config.py

```diff
@@ -31,28 +31,37 @@
 
 from __future__ import absolute_import
 
+from grpc.beta.interfaces import StatusCode
 from grpc.framework.interfaces.face import face
 from . import grpc
 
 
-# TODO: Determine which exceptions we support.
-EXCEPTION_NAMES = {
-    'ABORTED': face.AbortionError,
-    'CANCELLED': face.CancellationError,
-    'DATA_LOSS': face.NetworkError,
-    'DEADLINE_EXCEEDED': face.ExpirationError,
-    'FAILED_PRECONDITION': face.AbortionError,
-    'INTERNAL': face.AbortionError,
-    'INVALID_ARGUMENT': face.AbortionError,
-    'NOT_FOUND': face.AbortionError,
-    'OUT_OF_RANGE': face.AbortionError,
-    'PERMISSION_DENIED': face.AbortionError,
-    'RESOURCE_EXHAUSTED': face.AbortionError,
-    'UNAUTHENTICATED': face.AbortionError,
-    'UNAVAILABLE': face.AbortionError,
-    'UNIMPLEMENTED': face.AbortionError,
-    'UNKNOWN': face.AbortionError}
-
-
-create_stub = grpc.create_stub  # pylint: disable=invalid-name
+def exc_to_code(exc):
+    """Retrieves the status code from an exception"""
+    if not isinstance(exc, face.AbortionError):
+        return None
+    else:
+        return getattr(exc, 'code', None)
+
+
+STATUS_CODE_NAMES = {
```

Comment:
This is passed in to api_callable.construct_settings by the client. It is used to translate the strings that appear in the client config YAML into the actual objects representing the status codes. There will need to be a corresponding codegen update to rename `EXCEPTION_NAMES` to `STATUS_CODE_NAMES`.